include/cachemere/policy/eviction_lru.hpp

```diff
@@ -61,10 +61,15 @@ template<class Key, class Value> void EvictionLRU<Key, Value>::on_evict(const Ke
 {
     assert(!m_nodes.empty());
     assert(!m_keys.empty());
-    assert(m_keys.back().get() == key);
 
-    m_nodes.erase(key);
-    m_keys.pop_back();
+    if (m_keys.back().get() == key) {
```

Comment:
This change (along with the similar change in `eviction_segmented_lru.hpp`) is required because before implementing `remove()`, the only removals that could happen were on cache evict, which meant the evicted item was always at the tail of the list. Now that remove operations are random, we need to account for that.